e2e/components/button/button.e2e.ts

```diff
@@ -1,17 +1,23 @@
 import {browser, by, element} from 'protractor';
+import {screenshot} from '../../screenshot';
 
 describe('button', function () {
   describe('disabling behavior', function () {
     beforeEach(function() {
       browser.get('/button');
     });
     it('should prevent click handlers from executing when disabled', function () {
+      let testname =
```

Comment:
Jasmine 2.0 made it more difficult to get the current test name, but it's still possible by adding a custom reporter. [Example](https://github.com/jasmine/jasmine/issues/611#issuecomment-268504723).

Could could add a reporter that just updates a value which is then read in `screenshot.ts`